.github/workflows/ci.yml

```diff
@@ -0,0 +1,129 @@
+        if: ${{ failure() }}
+        run: golangci-lint run --timeout=5m -c .golangci.yaml
+
+  build:
+    runs-on: windows-2019
+    needs:
+      - test
+      - lint
+    env:
+      GO111MODULE: on
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          # fetch-depth required for gitversion in `Build` step
+          fetch-depth: 0
+
+      - name: Install Build deps
+        run: |
+          go get -u github.com/prometheus/promu@v0.11.1
+          go get -u github.com/josephspurrier/goversioninfo/cmd/goversioninfo@v1.2.0
+          # GOPATH\bin dir must be added to PATH else the `promu` and `goversioninfo` commands won't be found
+          echo "$(go env GOPATH)\bin" | Out-File -FilePath $env:GITHUB_PATH -Encoding utf8 -Append
+
+      - name: Build
+        run: |
+          # go mod download (or, if we don't call it, go build) will write every dependent package name to
+          # stderr, which will be interpreted as an error and abort the build if ErrorActionPreference is Stop,
+          # so we need to run it before setting the preference.
+          go mod download
+          $ErrorActionPreference = "Stop"
+
+          gitversion /output json /showvariable FullSemVer | Set-Content VERSION -PassThru
+          $Version = Get-Content VERSION
+          # Windows versioninfo resources need the file version by parts (but product version is free text)
+          $VersionParts = ($Version -replace '^v?([0-9\.]+).*$','$1').Split(".")
+          goversioninfo.exe -ver-major $VersionParts[0] -ver-minor $VersionParts[1] -ver-patch $VersionParts[2] -product-version $Version -platform-specific
+
+          make crossbuild
+          # GH requires all files to have different names, so add version/arch to differentiate
+          foreach($Arch in "amd64","386") {
+            Move-Item output\$Arch\windows_exporter.exe output\windows_exporter-$Version-$Arch.exe
+          }
+
+      - name: Upload Artifacts
+        uses: actions/upload-artifact@v2
+        with:
+          name: windows_exporter_binaries
+          path: output\windows_exporter-*.exe
+
+      - name: Build Release Artifacts
+        if: startsWith(github.ref, 'refs/tags/')
+        run: |
+          $ErrorActionPreference = "Stop"
+          $BuildVersion = Get-Content VERSION
+          $TagName = $env:GITHUB_REF -replace 'refs/tags/', ''
+          # The MSI version is not semver compliant, so just take the numerical parts
+          $MSIVersion = $TagName -replace '^v?([0-9\.]+).*$','$1'
+          foreach($Arch in "amd64","386") {
+            Write-Verbose "Building windows_exporter $MSIVersion msi for $Arch"
+            .\installer\build.ps1 -PathToExecutable .\output\windows_exporter-$BuildVersion-$Arch.exe -Version $MSIVersion -Arch "$Arch"
+            Move-Item installer\Output\windows_exporter-$MSIVersion-$Arch.msi output\
+          }
+
+          promu checksum output\
+
+      - name: Release
+        uses: softprops/action-gh-release@v1
```

Comment:
In the interest of keeping the number of external dependencies with write access to the repo low: Would it be hard to implement this ourselves as a `make` target (or just a script directly in this file)? I don't know what it conveniences it might include, but I see that `gh` (the Github CLI) is pre-installed on the image, so perhaps it's equivalent to `gh release upload <tag> output\windows_exporter-*.msi output\windows_exporter-*.exe`?

Reply:
I think I have it now, `gh release upload` has been used in the `Release` step.